### PR TITLE
Task/add orgId to endpoints

### PR DESCRIPTION
**Pull request by anzioka**
PR: add organizationId to endpoint paths to make chronicle work with apps v2

### CODE REVIEW

File: src/main/java/com/openlattice/chronicle/ChronicleApi.java
Comment (jokedurnez) [inline src/main/java/com/openlattice/chronicle/ChronicleApi.java:62]: This one os not deprecated right?

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (jokedurnez) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:132]: Does this need a V1-backwards-compatibility alternative?

Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:132]: No, it doesn't since it's not used directly by the android app

File: src/main/java/com/openlattice/chronicle/ChronicleApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleApi.java:62]: good catch!

**Comment by anzioka** [comment]
> Also wondering about this: https://github.com/openlattice/chronicle-api/blob/develop/chronicle.yaml
> 
> I've used this for myself, not sure if you want to maintain it. If so, then these endpoints should change as well. If not, maybe delete that file to avoid confusion.
> I also only superficially checked whether there are no overlapping paths, since a few changed. I'm assuming you also double checked that :)

@jokedurnez  What do you use the yaml file for?

**Review by jokedurnez — DISMISSED** [review]

**Review by makosblade — DISMISSED** [review]

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (katherinebernstein) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:194]: I think you'll need to keep `PREPROCESSED_PATH` as part of the path here, otherwise the controller won't be able to differentiate between an incoming request here vs to `getAllParticipantData` above

**Review by katherinebernstein — APPROVED** [review]

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:86]: At some point in the future, the "v1" endpoints won't exist and we'll be left with `methodV2` all over the place, and I don't think that's ideal. Furthermore, the actual API doesn't distinguish between v1 and v2, i.e. this is not a thing:

```
POST /v1/{study_id}/{participant_id}/{datasource_id}
POST /v2/{org_id}/{study_id}/{participant_id}/{datasource_id}
```

... so I'm not convinced that we should do this with the method names.

The API change we're making is that we're introducing endpoints that take an additional parameter to identify the organization. For all the `V2` methods, I prefer naming them something more semantically meaningful. For example:

```
// v1
UUID enrollSource(...)

// v2
UUID enrollDataSourceInOrgStudy(..)
// or
UUID orgStudyEnrollment(...)
// or
UUID registerForOrgStudy(...)
```

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:74]: Minor. Shouldn't it be `participant's data source` instead of `participant's data datasource`?

File: src/main/java/com/openlattice/chronicle/ChronicleApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleApi.java:63]: Same comment about the `V2`naming.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:133]: How come this method doesn't get a `V2` version? Actually, where is this being used?

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:160]: Just want to make the note... this is not backwards compatible. I don't think this is in use yet, so I don't think it'll be a problem.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:146]: Just want to make the note... this is not backwards compatible. I don't think this is in use yet, so I don't think it'll be a problem.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:230]: This looks like a breaking change.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:212]: This looks like a breaking change.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:246]: This looks like a breaking change.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:330]: This looks like a breaking change.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:315]: This looks like a breaking change.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:20]: This belongs with the other `*_PATH` constants below.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:30]: Please sort.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:49]: Please sort.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:42]: Please sort.

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:133]: At the moment it's not used by any client (web/android)

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:160]: Right!

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:230]: Correct. However, I removed the PARTICIPANT_PATH + DATA_PATH because they don't make sense being part of the the path, unlike in the endpoints that are for downloading participant's data (raw, preprocessed and usage). Also updated the web app since to use the corrected path.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:212]: https://github.com/openlattice/chronicle-api/pull/35#discussion_r485306813

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:315]: This doesn't need to be backward compatible since it's only used by the web app at the moment.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:212]: The only difference is ORGANIZATION_ID_PATH in the new path, and there is no need for backward compatibility

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:330]: https://github.com/openlattice/chronicle-api/pull/35#discussion_r485307956

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:246]: https://github.com/openlattice/chronicle-api/pull/35#discussion_r485306813

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (UnsungHero97) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:132]: Sooo... let's remove it?

**Review by geekbeast — APPROVED** [review]

**Comment by geekbeast** [comment]
> > Also wondering about this: https://github.com/openlattice/chronicle-api/blob/develop/chronicle.yaml
> > I've used this for myself, not sure if you want to maintain it. If so, then these endpoints should change as well. If not, maybe delete that file to avoid confusion.
> > I also only superficially checked whether there are no overlapping paths, since a few changed. I'm assuming you also double checked that :)
> 
> @jokedurnez What do you use the yaml file for?

Did you get an answer to this question?

**Comment by geekbeast** [comment]
So I don't have full context on this, but why are we adding organization id everywhere? I understand that it is needed for apps v2, but aren't we tracking this / looking this up internally in chronicle server in some fashion? 

Have we made sure that there will never be dual enrollment of a participant into two studies? Are we now storing an organization id on each phone and having to extend the deep link to contain an additional piece of information?

I'm especially curious about the write path modification-- I am not seeing how having an organization id is necessary.

**Review by geekbeast — COMMENTED** [review]
I need to understand what is going on a little better before I can approve this.

**Comment by anzioka** [comment]
> > > Also wondering about this: https://github.com/openlattice/chronicle-api/blob/develop/chronicle.yaml
> > > I've used this for myself, not sure if you want to maintain it. If so, then these endpoints should change as well. If not, maybe delete that file to avoid confusion.
> > > I also only superficially checked whether there are no overlapping paths, since a few changed. I'm assuming you also double checked that :)
> > 
> > 
> > @jokedurnez What do you use the yaml file for?
> 
> Did you get an answer to this question?

From Yoke:

> You can use such a file for a bunch of stuff.  For example, in the api-project, we have openlattice.yaml.  Based on that, we generate:
> - python and R clients (you can also generate javascript and android, but Hristo prefers using it himself): https://github.com/Lattice-Works/api-clients
> - This: https://docs.staging.openlattice.com/#/
> - I recently discovered drew is using it as well for insomnia
> I initially created it for chronicle just to be able to search around more easily, but I don’t think anyone else is using it.
> The file is the openapi-specification, it’s a really cool project.

**Comment by UnsungHero97** [comment]
@geekbeast...

> So I don't have full context on this, but why are we adding organization id everywhere?

The goal is to update the API for app configs v2 while being backwards compatible with existing installations of the Chronicle Android app. Perhaps using the organization id as part of the URL is not the best approach.

> I understand that it is needed for apps v2, but aren't we tracking this / looking this up internally in chronicle server in some fashion?

Chronicle server currently does not use an organization id for anything, so there's no logic for tracking or looking up organization ids.

> Have we made sure that there will never be dual enrollment of a participant into two studies?

No. It is theoretically possible for a participant to be enrolled in multiple studies at the same time. However, the Android app can only be enrolled in a single study at the moment.

> Are we now storing an organization id on each phone and having to extend the deep link to contain an additional piece of information?

Not yet. The Android app has not been changed yet. However, based on @anzioka's changes in this PR, it seems like we are heading in that direction.

File: src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java:132]: I think we should keep it. I have been using it **_heavily_** for sanity check.

Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java]: Removed

**Review by geekbeast — COMMENTED** [review]
I still think adding organization id to all the paths is redundant, especially since the backend should be able to look up the organization id based on the study id. It feels like we're adding an extra piece of data that needs to be moved around everywhere.

**Review by geekbeast — DISMISSED** [review]
Approved after meeting. Use of organization id is too streamline adoption of app configs.

**Review by UnsungHero97 — APPROVED** [review]

**Review by geekbeast — APPROVED** [review]

**Pull request merged**
